### PR TITLE
"vnext" -> "packages/react-native-windows"

### PR DESCRIPTION
**Pull request by NickGerleman**
1. Moves the directory
1. Add vnext to a graveyard section in gitignore to not show a whole bunch of untracked files on pull
1. Changes yarn workspaces config to remove vnext as a special package location 
1. Rename "vnext" -> "packages/react-native-windows" to try to catch config
1. Change backslash direction where misaligned with the rest of a path (search via regex)
1. Fix Metro Config, Flow Config, Just task, Layout-Headers, inside of package for relative paths with parent segments that have changed
1. Shorten "Microsoft.ReactNative.Managed.UnitTest" LKG file names which were hitting max path issues with Git
1. Revert references where vnext was not a path
1. Shorten excessively long RNW internal Appx target directory due to max path issues (#7906)

To do after merge:
1. Update docs site

This change conflicts with a lot of other work in the repo. Git is good at handling renames of existing edited files, showing a conflict, but allowing automatically moving during merge. Will want to merge this quickly after ready to avoid repeated conflicts though.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7902)

### CODE REVIEW

File: packages/e2e-test-app/windows/ReactUWPTestApp.sln [hunk 1]
```diff
@@ -26,24 +26,24 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "PropertySheets", "PropertyS
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ReactUWPTestApp", "ReactUWPTestApp\ReactUWPTestApp.csproj", "{ABBB0407-0E82-486F-94CE-710900FCAADC}"
 EndProject
-Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Common", "..\..\..\vnext\Common\Common.vcxproj", "{FCA38F3C-7C73-4C47-BE4E-32F77FA8538D}"
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "Common", "..\..\..\packages/react-native-windows\Common\Common.vcxproj", "{FCA38F3C-7C73-4C47-BE4E-32F77FA8538D}"
```
Comment (NickGerleman) [inline packages/e2e-test-app/windows/ReactUWPTestApp.sln:29]: Need to fixup for consistent slash direction

File: packages/react-native-windows/proposals/active/readme.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-This folder contains proposal documents for `react-native-windows @vnext` features that are under active development.
+This folder contains proposal documents for `react-native-windows @packages/react-native-windows` features that are under active development.
```
Comment (NickGerleman) [inline packages/react-native-windows/proposals/active/readme.md:1]: needs fix

File: packages/react-native-windows/CHANGELOG.json [hunk 3]
```diff
@@ -3064,7 +3064,7 @@
         ],
         "none": [
           {
-            "comment": "Update vnext/Microsoft.ReactNative.Cxx/README.md",
```
Comment (NickGerleman) [inline packages/react-native-windows/CHANGELOG.json:3067]: Need to revert changelog changes

**Comment by vmoroz** [comment]
It would be preferable that we do not need to go into the deep subfolder to the source files. It is better to use the 'src' folder instead of 'vnext' for the source files. The rest of the Node.js package definition can go into the packages/react-native-windows if needed. Having source files in the 'src' folder is the common open source practice. E.g. see https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/develop/data/spec.bs

**Review by vmoroz — CHANGES_REQUESTED** [review]
:clock1:

**Comment by NickGerleman** [comment]
Discussing offline, but I do not think such a convention is applicable when this is for existing code inside of a JS monorepo with multiple different packages (with some also having native code).

**Comment by NickGerleman** [comment]
That sort of structure would make sense inside of an individual package I think.

**Comment by NickGerleman** [comment]
Closing this pending further discussion.

**Pull request closed without merging**